matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/prune/RedactionEventProcessor.kt

```diff
@@ -167,4 +155,34 @@ internal class RedactionEventProcessor @Inject constructor() : EventInsertLivePr
             else -> emptyList()
         }
     }
+
+    private fun canPruneEventType(eventType: String): Boolean {
+        return when {
+            EventType.isCallEvent(eventType) -> false
+            EventType.isVerificationEvent(eventType) -> false
+            eventType == EventType.ROOM_KEY ||
+                    eventType == EventType.STATE_ROOM_WIDGET_LEGACY ||
+                    eventType == EventType.STATE_ROOM_WIDGET ||
+                    eventType == EventType.STATE_ROOM_NAME ||
+                    eventType == EventType.STATE_ROOM_TOPIC ||
+                    eventType == EventType.STATE_ROOM_AVATAR ||
+                    eventType == EventType.STATE_ROOM_THIRD_PARTY_INVITE ||
+                    eventType == EventType.STATE_ROOM_GUEST_ACCESS ||
+                    eventType == EventType.STATE_SPACE_CHILD ||
+                    eventType == EventType.STATE_SPACE_PARENT ||
+                    eventType == EventType.STATE_ROOM_TOMBSTONE ||
+                    eventType == EventType.STATE_ROOM_HISTORY_VISIBILITY ||
+                    eventType == EventType.STATE_ROOM_RELATED_GROUPS ||
+                    eventType == EventType.STATE_ROOM_PINNED_EVENT ||
+                    eventType == EventType.STATE_ROOM_ENCRYPTION ||
+                    eventType == EventType.STATE_ROOM_SERVER_ACL ||
+                    eventType == EventType.ROOM_KEY_REQUEST ||
+                    eventType == EventType.FORWARDED_ROOM_KEY ||
+                    eventType in EventType.ROOM_KEY_WITHHELD.values ||
+                    eventType == EventType.REQUEST_SECRET ||
+                    eventType == EventType.SEND_SECRET ||
+                    eventType == EventType.REACTION -> false
```

Comment:
As discussed with @BillCarsonFr some event types in this list are only toDevice events, so should never be used here. The fact that all Event types are defined in `EventType` does not help.

Not blocking though.

Reply:
I removed some to-device event types from the list, tell me if I am wrong https://github.com/vector-im/element-android/pull/7611/commits/57ef4fec12a4d07e6359e2fe3d8d52f09a44ace9

Reply:
Also I think the tombstone event can be redacted, in this case we should probably consider it? It can be done later as this is not a regression and common usecase.

Reply:
> I removed some to-device event types from the list, tell me if I am wrong [f59be73](https://github.com/vector-im/element-android/commit/f59be7334cd66f056dd791b2ee44a046257b5f9d)

This is LGTM, thanks!